pandas/core/reshape/reshape.py

```diff
@@ -1053,6 +1053,141 @@ def get_empty_frame(data) -> DataFrame:
         return DataFrame(dummy_mat, index=index, columns=dummy_cols)
 
 
+def from_dummies(
+    data: DataFrame,
+    to_series: bool = False,
+    prefix_sep: str | list[str] | dict[str, str] = "_",
+    columns: None | list[str] = None,
+    dropped_first: None | str | list[str] | dict[str, str] = None,
+) -> Series | DataFrame:
+    """
+    soon
+    """
+    from pandas.core.reshape.concat import concat
+
+    if data.isna().any().any():
+        raise ValueError(
+            f"Dummy DataFrame contains NA value in column: "
+            f"'{data.columns[data.isna().any().argmax()]}'"
+        )
+
+    if to_series:
+        return _from_dummies_1d(data, dropped_first)
+
+    data_to_decode: DataFrame
+    if columns is None:
+        # index data with a list of all columns that are dummies
+        cat_columns = []
+        non_cat_columns = []
+        for col in data.columns:
+            if any(ps in col for ps in prefix_sep):
+                cat_columns.append(col)
+            else:
+                non_cat_columns.append(col)
+        data_to_decode = data[cat_columns].astype("boolean")
+        non_cat_data = data[non_cat_columns]
+    elif not is_list_like(columns):
+        raise TypeError("Input must be a list-like for parameter 'columns'")
+    else:
+        data_to_decode = data[columns].astype("boolean")
+        non_cat_data = data[[col for col in data.columns if col not in columns]]
```

Comment:
I  just realized that this assumes that the non-dummy columns should be dropped from the return, adding an additional argument just for deciding if the non-dummies should be kept or not seems overkill. My idea would be to just return all columns specified and check if they need to be converted or not. So it is easy to pass a slice of input data columns.